### PR TITLE
BUG: cumulative functions with ea dtype not handling NA correctly and casting to object

### LINKED ISSUE #39479: BUG: cumsum() shows inconsistent results in nullable dtype
- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of pandas.
- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
# import pandas as pd
# import numpy as np

# like https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.cumsum.html
# Ex1
>>> pd.Series([2, np.nan, 5, -1, 0], dtype="float").cumsum()
0    2.0
1    NaN
2    7.0
3    6.0
4    6.0
dtype: float64

# Ex2
>>> pd.Series([2, np.nan, 5, -1, 0], dtype="Float64").cumsum()
0    2.0
1    2.0
2    7.0
3    6.0
4    6.0
dtype: object

# Ex3
>>> pd.Series([2, np.nan, 5, -1, 0], dtype="Int64").cumsum()
0       2
1    <NA>
2    <NA>
3    <NA>
4    <NA>
dtype: object
```

#### Problem description

- Ex1 is expected behavior while Ex2 and Ex3 yield different values.
- Ex2 returns `dtype: object` but `Float64` is expected, the same applies to Ex3.
- Other cum-family methods (cumprod, cummax, cummin) have the same problem.

#### Expected Output

```python
# Ex2
>>> pd.Series([2, np.nan, 5, -1, 0], dtype="Float64").cumsum()
0    2.0
1    <NA>
2    7.0
3    6.0
4    6.0
dtype: Float64

# Ex3
>>> pd.Series([2, np.nan, 5, -1, 0], dtype="Int64").cumsum()
0       2
1    <NA>
2       7
3       6
4       6
dtype: Int64
```

#### Environment

```Dockerfile
FROM python:3.9.1-slim-buster
WORKDIR /home
RUN pip install pandas==1.2.1
CMD ["/usr/local/bin/python"]
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 9d598a5e1eee26df95b3910e3f2934890d062caa
python           : 3.9.1.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.121-linuxkit
Version          : #1 SMP Tue Dec 1 17:50:32 UTC 2020
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.1
numpy            : 1.19.5
pytz             : 2020.5
dateutil         : 2.8.1
pip              : 21.0
setuptools       : 52.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #39479
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -1728,12 +1731,27 @@ def na_accum_func(values: ArrayLike, accum_func, *, skipna: bool) -> ArrayLike:
                 result, dtype=orig_dtype
             )
 
-    elif skipna and not issubclass(values.dtype.type, (np.integer, np.bool_)):
+    elif (
```
Comment (jreback) [inline pandas/core/nanops.py:1734]: i think maybe ok to have is_ea_dtype be a separate elif; much easier to rationalize here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/nanops.py (hunk 1 above)
Comment (phofl) [reply pandas/core/nanops.py]: Could do it like that

**Comment by jorisvandenbossche** [comment]
More general solution is being worked on in https://github.com/pandas-dev/pandas/pull/28509 (which doesn't mean a short term solution can't be merged, I didn't closely look at either PR, just noting)

File: pandas/core/nanops.py [hunk 2]
```diff
@@ -1728,6 +1729,21 @@ def na_accum_func(values: ArrayLike, accum_func, *, skipna: bool) -> ArrayLike:
...
+                np.minimum.accumulate: np.iinfo(values.dtype.type).max,
+            }[accum_func]
+
+        vals = values.copy()
+        mask = isna(vals)
+        mask_copy = np.copy(mask)
+        vals[mask] = mask_a
+        result = accum_func(vals, axis=0)
+        result[mask_copy] = mask_b
+        result = sanitize_array(result, None, values.dtype)
```
Comment (jbrockmendel) [inline pandas/core/nanops.py:1745]: sanitize_array is relatively heavy-weight.  what cases is it handling?

**Comment by simonjayhawkins** [comment]
> More general solution is being worked on in #28509 (which doesn't mean a short term solution can't be merged, I didn't closely look at either PR, just noting)

@phofl can you resolve conflicts

**Comment by simonjayhawkins** [comment]
@phofl closing as stale. https://github.com/pandas-dev/pandas/pull/39483#discussion_r593823587 needs addressing. reopen when ready.

**Pull request closed without merging**